src/contract/HandlerBasedContract.ts

```diff
@@ -525,15 +524,8 @@ export class HandlerBasedContract<State> implements Contract<State> {
         this._rootSortKey = sortedInteractions[sortedInteractions.length - 1].sortKey;
       }
     }
-    if (this.isRoot()) {
-      this._eoEvaluator = new EvaluationOptionsEvaluator(
-        this.evaluationOptions(),
-        contractDefinition.manifest?.evaluationOptions
-      );
-    }
-    const contractEvaluationOptions = this.isRoot()
-      ? this._eoEvaluator.rootOptions
-      : this.getEoEvaluator().forForeignContract(contractDefinition.manifest?.evaluationOptions);
+
+    const contractEvaluationOptions = this.resolveEvaluationOptions(contractDefinition.manifest?.evaluationOptions);
```

Comment:
hmm, isn't it already done at line 502? That one might be optional, so I would move the contractEvaluationOptions before the big fuckin' if and check here if it hasn't be already set.

Reply:
sure